File: app/src/main/java/Logger.java
import javax.swing.*;
import javax.swing.text.BadLocationException;
import javax.swing.text.SimpleAttributeSet;
import javax.swing.text.StyleConstants;

public class Logger {
    private final JTextPane textPanel;
    private final SimpleAttributeSet info = info();
    private final SimpleAttributeSet warn = warn();

    public Logger(JTextPane textPanel) {
        this.textPanel = textPanel;
    }

    public void info(String message) {
        try {
            textPanel.getStyledDocument().insertString(textPanel.getStyledDocument().getLength(), message + "\n", info);
            textPanel.setCaretPosition(textPanel.getDocument().getLength());
        } catch (BadLocationException e) {
            throw new RuntimeException(e);
        }
    }

    public void warn(String message) {
        try {
            textPanel.getStyledDocument().insertString(textPanel.getStyledDocument().getLength(), message + "\n", warn);
            textPanel.setCaretPosition(textPanel.getDocument().getLength());
        } catch (BadLocationException e) {
            throw new RuntimeException(e);
        }
    }

    private SimpleAttributeSet info() {
        return new SimpleAttributeSet();
    }

    private SimpleAttributeSet warn() {
        SimpleAttributeSet attributeSet = new SimpleAttributeSet();
        StyleConstants.setBold(attributeSet, true);

        return attributeSet;
    }
}

File: yagoc-desktop/src/main/kotlin/jenm/yagoc/ui/swing/Logger.kt
package jenm.yagoc.ui.swing

import javax.swing.JTextPane
import javax.swing.SwingUtilities
import javax.swing.text.SimpleAttributeSet

data class Logger(val textPanel: JTextPane) {
    private val infoAttributeSet = info()

    fun info(message: String) {
        textPanel.styledDocument.insertString(textPanel.styledDocument.length, message, infoAttributeSet)
        textPanel.caretPosition = textPanel.document.length
        SwingUtilities.invokeLater { textPanel.update(textPanel.graphics) }
    }

    fun debug(message: String) {
        println(message)
    }

    private fun info(): SimpleAttributeSet {
        return SimpleAttributeSet()
    }
}